src/reconcile/rec_write.c

```diff
@@ -1994,16 +2017,19 @@ __rec_split(WT_SESSION_IMPL *session, WT_RECONCILE *r, size_t next_len)
 		/* If the next object fits into this page, we're good to go. */
 		if (r->space_avail >= next_len)
 			return (0);
+		break;
```

Comment:
I've pushed a branch that's got a slightly different fix (I kept your idea of bringing back SPLIT_MAX, but added a `FALLTHROUGH` in the bottom of that case).